mailbox/opensearch/src/main/java/org/apache/james/mailbox/opensearch/events/OpenSearchListeningMessageSearchIndex.java

```diff
@@ -174,59 +174,84 @@ private Mono<Void> handleMailboxEvent(Event event, MailboxSession session, Mailb
 
     private Mono<Void> handleExpungedEvent(MailboxEvents.Expunged expunged, MailboxSession session, MailboxId mailboxId) {
         if (expunged.isMoved()) {
-            return Mono.empty();
+            return Mono.justOrEmpty(expunged.movedToMailboxId())
+                .flatMap(movedToMailboxId -> Flux.fromIterable(expunged.getExpunged().values())
+                    .concatMap(oldMessageData -> handleExpungedMessage(oldMessageData, mailboxId, movedToMailboxId, session))
+                    .then(Mono.defer(() -> delete(session, mailboxId, expunged.getUids()))));
         } else {
             return delete(session, mailboxId, expunged.getUids());
         }
     }
 
-    private Mono<Void> handleAddedEvent(MailboxEvents.Added event, MailboxSession session, MailboxEvents.Added addedEvent, MailboxId mailboxId) {
-        if (addedEvent.isMoved()) {
-            MailboxId movedFromMailboxId = addedEvent.movedFromMailboxId().get();
-            return Flux.fromIterable(addedEvent.getUids())
-                .concatMap(messageUid -> addByIndexedDocument(movedFromMailboxId, addedEvent.getMailboxId(), messageUid, session))
-                .onErrorResume(OpenSearchNotFoundException.class,
-                    notFoundException -> addByFetchMailboxMessage(event, session, mailboxId))
-                .thenEmpty(Mono.defer(() -> delete(session, movedFromMailboxId, addedEvent.getUids())));
-        } else {
-            return addByFetchMailboxMessage(event, session, mailboxId);
-        }
+    private Mono<Void> handleExpungedMessage(MessageMetaData expungedMessageMetaData,
+                                             MailboxId movedFromMailboxId,
+                                             MailboxId movedToMailboxId,
+                                             MailboxSession session) {
+        return newIndexByIndexedDocument(expungedMessageMetaData.getUid(), movedFromMailboxId, movedToMailboxId, session, expungedMessageMetaData.getMessageId())
+            .onErrorResume(OpenSearchNotFoundException.class,
+                notFoundException -> {
+                    LOGGER.warn("________Can not find message {} in mailbox {} for reindexing",
```

Comment:
Why `_______________` ?

Reply:
Ah, I forgot, I marked it for my track log, let me update it